matchzoo/embedding.py

```diff
@@ -1,143 +1,149 @@
 """Matchzoo toolkit for token embedding."""
-import numpy as np
-import logging
-from tqdm import tqdm
 
+import abc
+import csv
+import typing
 
-logger = logging.getLogger(__name__)
+import numpy as np
+import pandas as pd
 
 
-class Embedding(object):
+class Embedding(abc.ABC):
     """
     Embedding class.
-
-    TODO: https://github.com/dmlc/gluon-nlp/blob/
-          5e65c4751e1be8920b1021822a988af77224902f/
-          gluonnlp/embedding/token_embedding.py
-
-    Examples:
-        >>> import matchzoo as mz
-        >>> embed = mz.datasets.toy.load_embedding()
-        >>> # Need term_index
-        >>> embed.build({'G': 1, 'C': 2, 'D': 3, 'A': 4, '[PAD]': 0})
-        >>> index_state = embed.index_state
-        >>> index_state # doctest: +SKIP
-        {4: 1, 2: 1, 3: 1, 1: 2, 0: 0}
-        >>> embedding_mat = embed.embedding_mat
-        >>> embedding_mat # doctest: +SKIP
-        array([[ 0.       ,  0.       ,  0.       ,  0.       ,  0.        ,
-             0.        ,  0.        ,  0.        ,  0.        ,  0.        ],
-           [ 0.22785383, -0.09405118,  0.15446669,  0.20727136,  0.05943427,
-             0.12673594,  0.02788511, -0.0433806 , -0.20548974,  0.24532762],
-           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
-             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
-           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
-             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
-           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
-             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ]])
-        >>> embedding_mat[0] # doctest: +SKIP
-        array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
-
     """
 
-    def __init__(self, embedding_file):
-        """
-        Class initialization.
-
-        :param embedding_file: the embeding input file path.
-        """
-        self._embedding_file = embedding_file
-        self._embedding_dim = 0
-        self._index_state = {}
-        self._embedding_mat = None
-
+    def __init__(self):
+        """Init."""
+        self._vocab = {}
+        self._matrix = None
+        self._dimension = 0
```

Comment:
Is `_dimension` necessary? If you have `_matrix`, you always know the dimension right?

Reply:
yes